cmd/habitat/community/executor.go

```diff
@@ -1,54 +1,152 @@
 	"github.com/eagraf/habitat/cmd/habitat/procs"
-	"github.com/eagraf/habitat/pkg/compass"
+	"github.com/eagraf/habitat/structs/community"
 	"github.com/rs/zerolog/log"
 )
 
+type TransitionExecutor func(update *state.StateUpdate) error
+
 type CommunityExecutor struct {
-	processManager *procs.Manager
+	node *node.Node
 }
 
-func NewCommunityExecutor(p *procs.Manager) *CommunityExecutor {
+func NewCommunityExecutor(n *node.Node) *CommunityExecutor {
 	return &CommunityExecutor{
-		processManager: p,
+		node: n,
+	}
+}
+
+// GetTransitionExecutor maps transition types to executor functions
+func (e *CommunityExecutor) GetTransitionExecutor(transitionType string) TransitionExecutor {
+	switch transitionType {
+	case state.TransitionTypeInitializeIPFSSwarm:
+		return e.InitializeIPFSSwarm
+	case state.TransitionTypeStartProcess:
+		return e.StartProcess
+	case state.TransitionTypeStopProcess:
+		return e.StopProcess
+	case state.TransitionTypeStartProcessInstance:
+		return e.StartProcessInstance
+	case state.TransitionTypeStopProcessInstance:
+		return e.StartProcessInstance
+	default:
+		return nil
 	}
 }
 
 func (e *CommunityExecutor) Execute(update *state.StateUpdate) {
 	log.Info().Msgf("executing %s state transition", update.TransitionType)
-	newState, err := update.State()
+
+	transitionExecutor := e.GetTransitionExecutor(update.TransitionType)
+	if transitionExecutor != nil {
+		err := transitionExecutor(update)
+		if err != nil {
+			log.Error().Err(err).Msgf("error executing %s", update.TransitionType)
+		}
+	}
+}
+
+func (e *CommunityExecutor) InitializeIPFSSwarm(update *state.StateUpdate) error {
+	/*newState, err := update.State()
 	if err != nil {
-		log.Error().Msgf("error getting new state: %s", err)
+		return err
 	}
-	switch update.TransitionType {
-	case state.TransitionTypeInitializeIPFSSwarm:
-		communityIPFSConfig, err := json.Marshal(newState.IPFSConfig)
-		if err != nil {
-			log.Error().Msgf("error marshaling IPFS config")
+	communityIPFSConfig, err := json.Marshal(newState.IPFSConfig)
+	if err != nil {
+		return fmt.Errorf("error marshaling IPFS config: %s", err)
+	}
+
+	communityIPFSConfigB64 := base64.StdEncoding.EncodeToString(communityIPFSConfig)
+	if err != nil {
+		return fmt.Errorf("error base64 encoding IPFS config: %s", err)
+	}
+
+	ipfsPath := filepath.Join(compass.CommunitiesPath(), newState.CommunityID, "ipfs")
+	args := []string{ipfsPath}
+	flags := []string{"-c", communityIPFSConfigB64}
+
+	_, err = e.node.ProcessManager.StartProcessInstance("ipfs-driver", newState.CommunityID, args, []string{}, flags)
+	if err != nil {
+		return fmt.Errorf("error starting IPFS driver process: %s", err)
+	}*/
+	return nil
+}
+
+func (e *CommunityExecutor) StartProcess(update *state.StateUpdate) error {
+	return nil
+}
+
+func (e *CommunityExecutor) StopProcess(update *state.StateUpdate) error {
+	return nil
+}
+
+func (e *CommunityExecutor) StartProcessInstance(update *state.StateUpdate) error {
```

Comment:
what's supposed to be the difference between `StartProcess` and `StartProcessInstance`?

Reply:
A `Process` is a logical process that contains all the config, args etc used to run a process. `ProcessInstance` represents the mapping of a `Process` to a `Node`. We need this separation because we could have multiple instances of a `Process` running in the same community